crypto/src/merkle.rs

```diff
@@ -261,7 +261,7 @@ mod tests {
 
     fn test_hashes(n_hashes: u8) -> Vec<HashOf<()>> {
         (1..=n_hashes)
-            .map(|i| Hash::prehashed([i; Hash::LENGTH]).typed())
+            .map(|i| Hash::prehashed([i; Hash::LENGTH]).typed_unchecked())
```

Comment:
What's your opinion on marking functions like this as `unsafe`? I heard some people complain that it disables optimisations, but I found the opposite to be true, most of the time, Rust takes into account if you have unsafe blocks and it can eliminate more code if it can know that you can't create an invalid object. 

This comes from age-old C++ optimisations where LLVM tracks some invariants if you have a limited set of constructors.

Reply:
I agree with @mversic , `unsafe` should stay for memory and UB issues while business logic should be marked on another level. `unchecked` is fine for me